### PR TITLE
normalize sessionSpaceCompressedId to opSpace

### PR DESCRIPTION
**Pull request by daesunp**
## Description

This PR addresses a bug where the sessionSpaceCompressedIds were not being normalized to the op space during encoding.

### CODE REVIEW

File: packages/dds/tree/src/feature-libraries/chunked-forest/codec/chunkDecoding.ts [hunk 1]
```diff
@@ -38,15 +38,22 @@ import {
 	type EncodedValueShape,
 	SpecialField,
 } from "./format.js";
-import type { IIdCompressor, SessionSpaceCompressedId } from "@fluidframework/id-compressor";
+import type { IIdCompressor, OpSpaceCompressedId, SessionId } from "@fluidframework/id-compressor";
 
+export interface IdCompressorContext {
```
Comment (taylorsw04) [inline packages/dds/tree/src/feature-libraries/chunked-forest/codec/chunkDecoding.ts:43]: I'd rename this to IdDecodingContext, and I recommend documenting the orginator field, something like "the creator of any local IDs to be decoded"

File: packages/dds/tree/src/test/feature-libraries/chunked-forest/codec/schemaBasedEncoding.spec.ts [hunk 2]
```diff
@@ -306,9 +306,11 @@ describe("schemaBasedEncoding", () => {
 
 				const context: FieldBatchEncodingContext = {
 					encodeType: TreeCompressionStrategy.Compressed,
+					originatorId: testIdCompressor.localSessionId,
 					schema: { schema: storedSchema, policy: defaultSchemaPolicy },
 					idCompressor,
 				};
+				idCompressor.finalizeCreationRange(idCompressor.takeNextCreationRange());
```
Comment (taylorsw04) [inline packages/dds/tree/src/test/feature-libraries/chunked-forest/codec/schemaBasedEncoding.spec.ts:313]: are there any other snapshot tests that use identifiers that we should do something similar to?

File: packages/dds/tree/src/feature-libraries/chunked-forest/codec/chunkDecoding.ts (hunk 1 above)
Comment (daesunp) [reply packages/dds/tree/src/feature-libraries/chunked-forest/codec/chunkDecoding.ts]: Renamed interface to `IdDecodingContext` and added documentation for originatorId field

File: packages/dds/tree/src/feature-libraries/chunked-forest/codec/chunkDecoding.ts [hunk 3]
```diff
@@ -38,15 +38,25 @@ import {
...
  */
-export function decode(chunk: EncodedFieldBatch, idCompressor: IIdCompressor): TreeChunk[] {
+export function decode(
+	chunk: EncodedFieldBatch,
+	idCompressorContext: { idCompressor: IIdCompressor; originatorId: SessionId },
+): TreeChunk[] {
 	return genericDecode(
 		decoderLibrary,
-		new DecoderContext(chunk.identifiers, chunk.shapes, idCompressor),
+		new DecoderContext(chunk.identifiers, chunk.shapes, idCompressorContext),
```
Comment (taylorsw04) [inline packages/dds/tree/src/feature-libraries/chunked-forest/codec/chunkDecoding.ts:59]: I'd also do a find-replace for idCompressorContext -> idDecodingContext :)

File: packages/dds/tree/src/test/feature-libraries/chunked-forest/codec/schemaBasedEncoding.spec.ts (hunk 2 above)
Comment (daesunp) [reply packages/dds/tree/src/test/feature-libraries/chunked-forest/codec/schemaBasedEncoding.spec.ts:313]: The only other place where I think we could do something similar are the `"identifier field encoding"` tests in `chunkEncodingEndToEnd.ts`, which is currently using the `testIdCompressor` (which always generated finalized ids). Should these be updated to use `createIdCompressor()`, and call `finalizeCreationRange` instead?

File: packages/dds/tree/src/feature-libraries/chunked-forest/codec/chunkDecoding.ts (hunk 3 above)
Comment (daesunp) [reply packages/dds/tree/src/feature-libraries/chunked-forest/codec/chunkDecoding.ts]: renamed variable to `idDecodingContext`

File: packages/dds/tree/src/test/feature-libraries/chunked-forest/codec/schemaBasedEncoding.spec.ts (hunk 2 above)
Comment (taylorsw04) [reply packages/dds/tree/src/test/feature-libraries/chunked-forest/codec/schemaBasedEncoding.spec.ts:313]: No, I think this is fine. Let's get this in.

**Comment by taylorsw04** [comment]
@daesunp looks like you forgot to check in the snapshot files for your new tests. I'll approve and we can merge once those are in.

**Review by taylorsw04 — APPROVED** [review]

**Pull request merged**